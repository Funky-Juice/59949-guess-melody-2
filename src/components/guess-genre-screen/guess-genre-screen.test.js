import renderer from 'react-test-renderer';
import GuessGenreScreen from './guess-genre-screen';

const mockQuestion = {
  type: `genre`,
  genre: `jazz`,
  answers: [
    {
      src: `https://upload.wikimedia.org/wikipedia/commons/3/34/Metalmania_(ISRC_USUAN1700023).mp3`,
      genre: `rock`
    },
    {
      src: `https://upload.wikimedia.org/wikipedia/commons/5/5d/Acid_Trumpet_%28ISRC_USUAN1100339%29.mp3`,
      genre: `jazz`
    },
    {
      src: `https://upload.wikimedia.org/wikipedia/commons/d/db/Keef_Trouble%2C_Oasis.oga`,
      genre: `pop`
    },
    {
      src: `https://upload.wikimedia.org/wikipedia/commons/c/c2/Synth_pop_with_4_on_the_floor.ogg`,
      genre: `electronic`
    }
  ]
};


it(`GuessGenreScreen correctly renders`, () => {
  const tree = renderer
    .create(<GuessGenreScreen question={mockQuestion} time={0} errors={0} screenIndex={0} onAnswer={() => {}}/>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
