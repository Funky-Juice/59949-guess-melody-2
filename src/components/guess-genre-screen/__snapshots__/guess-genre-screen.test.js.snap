// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GuessGenreScreen correctly renders 1`] = `
<section
  className="game__screen"
>
  <h2
    className="game__title"
  >
    Выберите 
    jazz
     треки
  </h2>
  <form
    className="game__tracks"
    onSubmit={[Function]}
  >
    <div
      className="track"
    >
      <div
        className="game__answer"
      >
        <input
          className="game__input visually-hidden"
          id="answer-1"
          name="answer"
          type="checkbox"
          value="rock"
        />
        <label
          className="game__check"
          htmlFor="answer-1"
        >
          Отметить
        </label>
      </div>
    </div>
    <div
      className="track"
    >
      <div
        className="game__answer"
      >
        <input
          className="game__input visually-hidden"
          id="answer-2"
          name="answer"
          type="checkbox"
          value="jazz"
        />
        <label
          className="game__check"
          htmlFor="answer-2"
        >
          Отметить
        </label>
      </div>
    </div>
    <div
      className="track"
    >
      <div
        className="game__answer"
      >
        <input
          className="game__input visually-hidden"
          id="answer-3"
          name="answer"
          type="checkbox"
          value="pop"
        />
        <label
          className="game__check"
          htmlFor="answer-3"
        >
          Отметить
        </label>
      </div>
    </div>
    <div
      className="track"
    >
      <div
        className="game__answer"
      >
        <input
          className="game__input visually-hidden"
          id="answer-4"
          name="answer"
          type="checkbox"
          value="electronic"
        />
        <label
          className="game__check"
          htmlFor="answer-4"
        >
          Отметить
        </label>
      </div>
    </div>
    <button
      className="game__submit button"
      type="submit"
    >
      Ответить
    </button>
  </form>
</section>
`;
